it/src/main/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManager.java

```diff
@@ -90,6 +91,9 @@ private DefaultSpannerResourceManager(Builder builder) {
     this.spanner = spanner;
     this.instanceAdminClient = spanner.getInstanceAdminClient();
     this.databaseAdminClient = spanner.getDatabaseAdminClient();
+
+    // Check that the project ID conforms to GCS standards
+    checkValidProjectId(projectId);
```

Comment:
Maybe check this at the beginning of the method? This way we can fail fast before creating the clients

Reply:
Fixed in Spanner and Bigtable constructors. Thanks.